e2e/tests/session_steps_test.go

```diff
@@ -42,7 +42,21 @@ const (
 )
 
 func (s *suite) TheUserShouldWaitForTheModuleMessageToBeSubmitted(module, msgType string) {
-	s.waitForTxResultEvent(newEventMsgTypeMatchFn(module, msgType))
+	event := s.waitForTxResultEvent(newEventMsgTypeMatchFn(module, msgType))
+
+	// Currently the proof submission fee may be higher than the reward amount,
+	// an adjustment the current balance to account for the proof submission fee
+	// is needed so validateAmountChange can assert on the reward change with the
+	// given condition.
+	if msgType == "SubmitProof" {
```

Comment:
I see what you're doing here, but this is a "hidden side effect".

I think you should just either:
1. Add a step call `EnsureBalanceIsHighEnoughForProofSubmission`
OR
2. Update the default configs of LocalNet to give everyone high enough balances that this doesn't matter

My preference is (2).

Reply:
This is actually not about having enough funds.

It is to make the step `Then the account balance of "supplier1" should be "420" uPOKT "more" than before` succeed without making it more complicated
* `fee > reward` -> `lower balance`
* `proof not required` -> `no fee`
* step needing to know about previous ones
* ...

There's actually an elegant solution without side effect.
Take a snapshot of the balance after the proof submission.